### PR TITLE
quote Windows arguments based on delimeters

### LINKED ISSUE #459: Options are quoted incorrectly on Windows
[Example CI](https://github.com/CharlesStover/monorepo-template/runs/4329442083?check_suite_focus=true#step:6:111)

Relevant error log:
```sh
ENOENT: no such file or directory, access 'D:\a\monorepo-template\monorepo-template\"packages\application"'
Error: ENOENT: no such file or directory, access 'D:\a\monorepo-template\monorepo-template\"packages\application"'
```

Problem: the `project` (value: `packages/application`) is quoted when used (`"packages/application"`), which isn't a valid file path.

[Likely problematic line](https://github.com/cypress-io/github-action/blob/master/index.js#L534): `cmd.push(quoteArgument(project))`
Speculation: The issue may be when [executing Cypress with `cmd`](https://github.com/CharlesStover/monorepo-template/runs/4329442083?check_suite_focus=true#step:6:112), where all the quotes getting quoted again?

I understand the problem solve of quoting parameters on Windows, but I'm not sure blanket-quoting all parameters is the right choice, especially given this side effect. I expect values without spaces to not be quoted, as having passed `packages/application` without the quotes would have worked fine, to my knowledge.

EDIT:
I've [turned debug on for Cypress](https://github.com/CharlesStover/monorepo-template/runs/4330484253?check_suite_focus=true#step:6:236) for some more context:
```sh
Cypress test command: npx yarn dlx cypress run --record --parallel --group "windows-edge" --tag "push" --project "packages/application" --config-file "cypress.json" --ci-build-id "push-39f6a232c4eeb4ebd417669f22e88de3c6e5ae4f" --browser edge

C:\Windows\system32\cmd.exe /D /S /C "C:\hostedtoolcache\windows\node\16.13.0\x64\npx.cmd yarn dlx cypress run --record --parallel --group """windows-edge""" --tag """push""" --project """packages/application""" --config-file """cypress.json""" --ci-build-id """push-39f6a232c4eeb4ebd417669f22e88de3c6e5ae4f""" --browser edge"

2021-11-26T05:05:57.449Z cypress:cli:cli cli starts with arguments ["C:\\hostedtoolcache\\windows\\node\\16.13.0\\x64\\node.exe","C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\xfs-d3bafd55\\dlx-4676\\.yarn\\unplugged\\cypress-npm-9.1.0-0dc889cc56\\node_modules\\cypress\\bin\\cypress","run","--record","--parallel","--group","\"windows-edge\"","--tag","\"push\"","--project","\"packages/application\"","--config-file","\"cypress.json\"","--ci-build-id","\"push-39f6a232c4eeb4ebd417669f22e88de3c6e5ae4f\"","--browser","edge"]
```
Note that the arguments here contain quotes when they should not.

### PR DESCRIPTION
**Pull request by CharlesStover**
Only quotes arguments on Windows when they contain [argument delimiters](https://ss64.com/nt/syntax-esc.html).

resolves #459

### CODE REVIEW

**Comment by CharlesStover** [comment]
@bahmutov I'm not sure who owns this repo. The PRs look pretty backed up, and this Windows issue is blocking my project, so `@` in hopes this reaches a maintainer.

**Comment by bahmutov** [comment]
I feel the pain, but I will leave it to the Cypress team to verify the solution and release it

**Review by mjhenkes — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**
